stackslib/src/net/api/tests/mod.rs

```diff
@@ -1420,3 +1424,32 @@ fn prefixed_hex_serialization() {
         assert_eq!(hex_str, format!("\"0x{}\"", to_hex(&inp.0)));
     }
 }
+
+#[test]
+fn test_successfull_resposes_preambles_have_canonical_stacks_tip_height() {
+    let addr: PeerHost = SocketAddr::new(IpAddr::V4(Ipv4Addr::LOCALHOST), 33333).into();
+    let test_observer = TestEventObserver::new();
+    let rpc_test = TestRPC::setup(function_name!());
+
+    let expected_height = rpc_test.peer_2.network.stacks_tip.height;
+
+    // Just a few different types of requests.
```

Comment:
You might consider using `rstest` here to synthesize tests for each and every request.